System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamFactoryTests.cs

```diff
@@ -44,6 +44,34 @@ public void MockFileStreamFactory_CreateForNonExistingFile_ShouldReturnStream(Fi
             Assert.IsNotNull(result);
         }
 
+        [Test]
+        [TestCase(FileMode.Create)]
+        [TestCase(FileMode.CreateNew)]
+        public void TW___MockFileStreamFactory_CreateForAnExistingFile_ShouldTruncateExistingFile(FileMode fileMode)
+        {
+            var fileSystem = new MockFileSystem();
+            string FilePath = XFS.Path("C:\\File.txt");
+
+            using(var stream = fileSystem.FileStream.Create(FilePath, fileMode, System.IO.FileAccess.Write))
+            {
+                using(var writer = new System.IO.StreamWriter(stream, System.Text.Encoding.UTF8, 4096, leaveOpen: true))
```

Comment:
There are problems in [appveyor ](https://ci.appveyor.com/project/tathamoddie/system-io-abstractions/builds/24463776/job/cu66awyi0m45so8d) tests because [NET40](https://docs.microsoft.com/pl-pl/dotnet/api/system.io.streamwriter.-ctor?view=netframework-4.0#System_IO_StreamWriter__ctor_System_String_System_Boolean_System_Text_Encoding_System_Int32_) didn't have constructor that support leaveOpen parameter.
To fix that you can make conditional scenario for `#if NET40` to get same react just don't create writer in using so [Dispose](https://stackoverflow.com/questions/2666888/is-there-any-way-to-close-a-streamwriter-without-closing-its-basestream) will not be run.

But please someone confirm that what i write make sense.

Reply:
I’d suggest just removing this argument, it does not seem to be necessary here. Same for the buffer size.

Reply:
Or even get rid of the StreamWriter and just write directly to the Stream.

Reply:
Updated... :)